analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -0,0 +1,194 @@
+            )
+        }
+
+    override fun resolveDependencies(definitionFile: File): List<ProjectAnalyzerResult> {
+        val workingDir = definitionFile.parentFile
+        val spdxDocument = SpdxModelMapper.read(definitionFile, SpdxDocument::class.java)
+
+        val projectPackage = spdxDocument.packages.singleOrNull { it.packageFilename.isEmpty() }
+
+        val project = if (projectPackage != null) {
+            val scopes = listOfNotNull(
+                Scope(
+                    name = DEFAULT_SCOPE_NAME,
+                    dependencies = getDependencies(projectPackage, spdxDocument)
+                ),
+                createScope(spdxDocument, projectPackage, SpdxRelationship.Type.BUILD_DEPENDENCY_OF),
+                createScope(spdxDocument, projectPackage, SpdxRelationship.Type.DEV_DEPENDENCY_OF),
+                createScope(spdxDocument, projectPackage, SpdxRelationship.Type.OPTIONAL_DEPENDENCY_OF),
+                createScope(spdxDocument, projectPackage, SpdxRelationship.Type.PROVIDED_DEPENDENCY_OF),
+                createScope(spdxDocument, projectPackage, SpdxRelationship.Type.RUNTIME_DEPENDENCY_OF),
+                createScope(spdxDocument, projectPackage, SpdxRelationship.Type.TEST_DEPENDENCY_OF)
+            ).toSortedSet()
+
+            Project(
+                id = projectPackage.toIdentifier(),
+                definitionFilePath = VersionControlSystem.getPathInfo(definitionFile).path,
+                declaredLicenses = sortedSetOf(projectPackage.licenseDeclared),
+                vcs = VcsInfo.EMPTY,
+                vcsProcessed = processProjectVcs(workingDir, VcsInfo.EMPTY, projectPackage.homepage),
+                homepageUrl = projectPackage.homepage,
+                scopes = scopes
+            )
+        } else {
+            // TODO: How to deals with relationships between SPDX packages if we do not have a project to attach
+            //       dependencies to?
+            Project.EMPTY
+        }
+
+        val nonProjectPackages = if (projectPackage != null) {
+            spdxDocument.packages - projectPackage
+        } else {
+            spdxDocument.packages
+        }
+
+        val packages = nonProjectPackages.mapTo(sortedSetOf()) { pkg ->
+            val packageDescription = pkg.description.takeUnless { it.isEmpty() }
+                ?: pkg.summary.takeUnless { it.isEmpty() }
+                ?: pkg.comment
+            val packageDir = workingDir.resolve(pkg.packageFilename)
+
+            Package(
+                id = pkg.toIdentifier(),
+                declaredLicenses = sortedSetOf(pkg.licenseDeclared),
+                concludedLicense = pkg.licenseConcluded.toSpdx(),
+                description = packageDescription,
+                homepageUrl = pkg.homepage,
+                binaryArtifact = RemoteArtifact.EMPTY, // TODO: Use "externalRefs"?
+                sourceArtifact = RemoteArtifact.EMPTY, // TODO: Use "externalRefs" or "sourceInfo"?
```

Comment:
Hint: in the reporter we settled for a separate package entry for each, binaryArtifact, sourceArtifact and vcs - whereas only the binary one is in the dependency tree, and then there are two GENERATED_FROM relationships.

Reply:
Oh. It sounds overly complicated to me to have multiple SpdxPackages for one ORT Package, esp. as you need to repeat all mandatory fields and they differ only in the `downloadLocation`. IMO we should read (and write) `externalRefs`, esp. as these already have package manager specific types.